auto-value-gson-annotations/src/main/java/com/ryanharter/auto/value/gson/GenerateTypeAdapter.java

```diff
@@ -0,0 +1,103 @@
+package com.ryanharter.auto.value.gson;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Annotate a given AutoValue-annotated type to indicate to auto-value-gson to generate its
+ * TypeAdapter in a separate class rather than an inner class of an intermediate AutoValue-generated
+ * class hierarchy.
+ */
+@Inherited
+@Retention(RUNTIME)
+@Target(TYPE)
+public @interface GenerateTypeAdapter {
+
+  TypeAdapterFactory FACTORY = new TypeAdapterFactory() {
+    private final Map<Class<?>, Constructor<? extends TypeAdapter>> adapters =
```

Comment:
Needs to be `ConcurrentHashMap` since this isn't a thread-safe type

Reply:
06ac048